### DISCUSSION TITLE
Upgrading to GORM V2

### DISCUSSION THREAD

**Discussion by glennzw** [gophish/gophish#2573] [Ideas]
Gophish uses gorm version [1.9.12](https://github.com/jinzhu/gorm/releases/tag/v1.9.12) with release date Dec 2019. Based on discussions with @edermi  in https://github.com/gophish/gophish/pull/2528 it might be worth upgrading to [gorm V2](https://github.com/go-gorm/gorm). In the release notes there are several [Breaking Changes](https://gorm.io/docs/v2_release_note.html#Breaking-Changes) that we'd need to be mindful of.

I'm opening this discussion thread for the community to have their input and share their experience as we explore the benefits and drawbacks of making this migration.

**Comment by edermi** [comment]
I'd like to bring the Postgresql topic on the table. We'd prefer to use it internally. From what I saw, https://github.com/gophish/gophish/blob/master/models/models.go is the only place where a differentiation between databases is made, and mostly because of database connection handling. 
Goose as well as Gorm support Postgres, so I don't expect it to be something requiring much maintenance on our side.

**Comment by troplolBE** [comment]
I think we could use an upgrade of Gorm. I see in the release notes that there are performance improvements and I think it would be a good thing to check if this is really the case. GoPhish is sometimes really struggling with speed and sometimes even fails to show landing pages.

Maybe using PostgresSQL as suggested by @edermi is a good alternative to the current sqlite. I also agree that removing goose out of the the project is a good thing. The less dependencies the better for future development.
